frame/support/src/traits/storage.rs

```diff
@@ -126,24 +126,27 @@ macro_rules! impl_incrementable {
 	($($type:ty),+) => {
 		$(
 			impl Incrementable for $type {
-				fn increment(&self) -> Self {
+				fn increment(&self) -> Option<Self> {
```

Comment:
nit: I think `Result<Self, ()>` conveys the fallibility better.

Reply:
I quickly looked into moving to Result and I like the Option better as it works better together with the `OptionQuery` of `NextCollectionId` and `PoolAssetId`. We have to do a lot of transforming of Option <-> Result otherwise.